### PR TITLE
Create Mean Expected Outcome Error Estimator

### LINKED ISSUE #8: Create the Mean Expected Outcome Error Estimator
Make it a child of the `ErrorEstimator` class.

See section 4.5 (i) in the paper.

### PR DESCRIPTION
**Pull request by andystanley**
Closes #8

### CODE REVIEW

File: mcos/mcos.py [hunk 1]
```diff
@@ -28,3 +29,11 @@ def simulate_optimizations(obs_simulator: ObservationSimulator, n_sims: int, opt
             'stdev': np.std(error_estimator[optimizers.name])
         } for optimizer in optimizers
     ]).set_index('optimizer')
+
+
+def optimal_allocation(mu: np.array, cov: np.array):
```
Comment (moneygeek) [inline mcos/mcos.py]: We should remove this. This is just one type of portfolio optimization. It does the same thing as `optimizer.allocate`

Comment (moneygeek) [reply mcos/mcos.py]: The "optimal" allocation should be obtained by taking running the optimizer using the original covariance matrix

Comment (andystanley) [reply mcos/mcos.py]: Aren't I already doing this here `optimizer.allocate(mu_hat, cov_hat)`? I'm not understanding the difference between calculating the allocation and "optimal" allocation.

File: mcos/error_estimator/expected_outcome.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import numpy as np
+
+from mcos.error_estimator import AbstractErrorEstimator
+
+
+class ExpectedOutcomeErrorEstimator(AbstractErrorEstimator):
+    """Error Estimator that calculates the mean difference in expected outcomes"""
+
+    def estimate(self, allocation: np.array, optimal_allocation: np.array) -> float:
+        return (optimal_allocation - allocation).mean()
```
Comment (andystanley) [inline mcos/error_estimator/expected_outcome.py]: This method implements the formula as defined by the paper:
Screen Shot 2020-01-27 at 8 35 41 PM
where the first value is the the optimal allocation from the original mu/cov pair and the second value is the allocation from a simulated mu/cov pair.

Comment (moneygeek) [reply mcos/error_estimator/expected_outcome.py]: You have the (w - what), but you're missing the mu part. The original mu has to be part of the equation. It should be `np.dot(optimal_allocation - allocation, mu)`

File: mcos/error_estimator/base.py [hunk 3]
```diff
@@ -2,13 +2,16 @@
 import numpy as np
 
 
-class ErrorEstimator(ABC):
+class AbstractErrorEstimator(ABC):
     @abstractmethod
-    def estimate(self, allocation: np.array) -> float:
+    def estimate(self, mu: np.array, cov: np.array, allocation: np.array, optimal_allocation: np.array) -> float:
```
Comment (andystanley) [inline mcos/error_estimator/base.py]: It looks like some of the other estimation methods require the covariance matrix, so I've added it as a parameter.

File: mcos/error_estimator/expected_outcome.py (hunk 2 above)
Comment (andystanley) [reply mcos/error_estimator/expected_outcome.py]: Updated this

**Pull request merged**